internal/controller/apiextensions/claim/configurator.go

```diff
@@ -92,7 +94,22 @@ func (c *APIDryRunCompositeConfigurator) Configure(ctx context.Context, cm resou
 	// external name.
 	en := meta.GetExternalName(ucp)
 
-	meta.AddAnnotations(ucp, ucm.GetAnnotations())
+	// Do not propagate *.kubernetes.io annotations down to the composite
+	// For example: when a claim gets deployed using kubectl,
+	// its kubectl.kubernetes.io/last-applied-configuration annotation
+	// should not be propagated to the corresponding composite resource,
+	// because:
+	// * XR was not created using kubectl
+	// * The content of the annotaton refers to the claim, not XR
+	// See https://kubernetes.io/docs/reference/labels-annotations-taints/
+	// for all annotations and their semantic
+	annot := ucm.GetAnnotations()
```

Comment:
nit: we can use a shorter name here per https://github.com/crossplane/crossplane/tree/master/contributing#use-descriptive-variable-names-sparingly

```suggestion
	a := ucm.GetAnnotations()
```